src/vos/vos_size.c

```diff
@@ -59,183 +59,168 @@ static int alloc_overhead = 16;
 	} while (0);
 
 #define PRINT_DYNAMIC(name, type, feats)				\
-	print_dynamic(fp, #name, &name);
+	print_dynamic(buf, #name, &name);
 
 #define PRINT_RECORD(name, type, feats)					\
-	print_record(fp, #name, &name);
+	print_record(buf, #name, &name);
+
+struct str_buffer {
+	unsigned int status;
+	size_t str_size;
+	size_t buf_size;
+	char *str;
+};
+
+static int
```

Comment:
Just a generic comment...this functionality would be generically useful and could live in gurt

Reply:
I created [DAOS-5360](https://jira.hpdd.intel.com/browse/DAOS-5360) to address it